pkg/migrate/cstor/cspc_generator.go

```diff
@@ -0,0 +1,187 @@
+/*
+Copyright 2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package migrate
+
+import (
+	"time"
+
+	cstor "github.com/openebs/api/pkg/apis/cstor/v1"
+	"github.com/openebs/api/pkg/apis/types"
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	csp "github.com/openebs/maya/pkg/cstor/pool/v1alpha3"
+	deploy "github.com/openebs/maya/pkg/kubernetes/deployment/appsv1/v1alpha1"
+	"github.com/openebs/maya/pkg/util/retry"
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	typeMap = map[string]string{
+		string(apis.PoolTypeStripedCPV):  string(apis.PoolStriped),
+		string(apis.PoolTypeMirroredCPV): string(apis.PoolMirrored),
+		string(apis.PoolTypeRaidzCPV):    string(apis.PoolRaidz),
+		string(apis.PoolTypeRaidz2CPV):   string(apis.PoolRaidz2),
+	}
+)
+
+func getBDList(cspObj apis.CStorPool) []cstor.CStorPoolInstanceBlockDevice {
+	list := []cstor.CStorPoolInstanceBlockDevice{}
+	for _, bdcObj := range cspObj.Spec.Group[0].Item {
+		list = append(list,
+			cstor.CStorPoolInstanceBlockDevice{
+				BlockDeviceName: bdcObj.Name,
+			},
+		)
+	}
+	return list
+}
+
+func getCSPCSpecForSPC(spc *apis.StoragePoolClaim, openebsNamespace string) (*cstor.CStorPoolCluster, error) {
+	cspClient := csp.KubeClient()
+	cspList, err := cspClient.List(metav1.ListOptions{
+		LabelSelector: string(apis.StoragePoolClaimCPK) + "=" + spc.Name,
+	})
+	if err != nil {
+		return nil, err
+	}
+	cspcObj := &cstor.CStorPoolCluster{}
+	cspcObj.Name = spc.Name
+	cspcObj.Annotations = map[string]string{
+		// This label will be used to disable reconciliation on the dependants.
+		// In this case that will be CSPI
```

Comment:
n_p: 
`// In this case that will be CSPI` --> `// In this case, it will be CSPI`